state-sync/aptos-data-client/src/peer_states.rs

```diff
@@ -13,7 +14,17 @@ use aptos_storage_service_types::{
 };
 use aptos_time_service::TimeService;
 use dashmap::DashMap;
-use std::{cmp::min, collections::HashSet, sync::Arc};
+use std::{
+    cmp::min,
+    collections::{BTreeMap, HashSet},
+    sync::Arc,
+    time::Duration,
+};
+
+// Useful constants
+const LOGS_FREQUENCY_SECS: u64 = 120; // 2 minutes
+const METRICS_FREQUENCY_SECS: u64 = 15; // 15 seconds
+const NUM_PEER_BUCKETS_FOR_METRICS: u8 = 5; // To avoid metric explosion, we bucket peers into groups
```

Comment:
should we do a power of 2 because you use the first byte module this value as bucket_id? I guess 16 is probably a proper choice.

Reply:
Hmm... I think 16 is too large (it'll still lead to some explosion). So, I've changed it to 4. 😄